### ISSUE TITLE
[BUG Bash] create_or_update method throws 400 error when using a valid subscription_id

### ISSUE THREAD

**Issue by lcouzens** [Azure/azure-sdk-for-python#21900]
**Describe the bug**
When trying to use the SDK to create a billing export via Create Or Update method it seems to throw a 400 error when using a valid subscription_id. Looking at the traceback it looks as if its some how setting the subscription_id to `providers`. My only guess for that is something is pursing out the ID from the URL but pulling the wrong part.

***Exception or Stack Trace***
```
 if response.status_code not in [200, 201]:
>           raise models.ErrorResponseException(self._deserialize, response)
E           azure.mgmt.costmanagement.models._models_py3.ErrorResponseException: (400) ServiceToServiceAuthorizationProvider.IsRPRegistered: {"error":{"code":"InvalidSubscriptionId","message":"The provided subscription identifier 'providers' is malformed or invalid."}}
```

**To Reproduce**
- Get CostManagementClient session
- Set scope to "/subscriptions/{subscription_id}/"
- Set export name
- Set required parameters for export
- Run costmgmt_client.exports.create_or_update(scope, export_name, parameters)

***Code Snippet***
```
scope = f"/subscriptions/{subscription_id}/"
    costmgmt_client = CostManagementClient(creds, subscription_id)
    parameters = {
        "deliveryInfo": {
            "destination": {
                "container": "cost",
                "rootFolderPath": "exports",
                "resourceId": "costmgmttest",
            },
        },
        "definition": {
            "dataSet": {"granularity": "daily"},
            "timeframe": "MonthToDate",
            "type": "ActualCost"
        }
    }
    costmgmt_client.exports.create_or_update(scope, "testexport", parameters)
```

**Expected behavior**
Billing export to be created correctly 

**Screenshots**

**Setup (please complete the following information):**
 - Python Version: Python 3.9
 - SDK Version: [azure-mgmt-costmanagement  0.2.0]

**Additional context**
Side note, if I pass in an invalid subscription_id I get the following which is expected:

```
if response.status_code not in [200, 201]:
>           raise models.ErrorResponseException(self._deserialize, response)
E           azure.mgmt.costmanagement.models._models_py3.ErrorResponseException: (SubscriptionNotFound) The subscription '658af4bc-4d35-4106-8c69-521552ee6fb3' could not be found.
```

Also another important thing to note here is I actually tried testing this out with the Docs code try session [0] and this fails in the same way, so I actually think this is possibly a bug in the API rather than the SDK. I'm not really sure how/where to raise that or test it further. 
It's almost like the API is pulling the `providers` string from the URL instead of the subscription_id.

[0] - https://docs.microsoft.com/en-us/rest/api/cost-management/exports/create-or-update#code-try-0

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by ashish-gupta1-by** [comment]
I am also facing the same issue. Not able to create adf. I am able to perform same activity by UI

**Comment by BigCat20196** [comment]
hi @lcouzens & @ashish-gupta1-by thanks for your feedback!

Now `azure-mgmt-costmanagement` has been released the latest version of 3.0 0, could you pls update the package with `pip install -U azure-mgmt-costmanagement` command and try again?
